sklearn/utils/tests/test_extmath.py

```diff
@@ -161,6 +161,26 @@ def test_randomized_svd_low_rank_all_dtypes(dtype):
     check_randomized_svd_low_rank(dtype)
 
 
+@pytest.mark.parametrize('dtype',
+                         (np.int32, np.int64, np.float32, np.float64))
+def test_randomized_eigsh(dtype):
+    """Test that `_randomized_eigsh` returns the appropriate components"""
+
+    X = np.diag(np.array([1., -2., 0., 3.], dtype=dtype))
+
+    # with 'module' selection method, the negative eigenvalue shows up
+    lambd_, alph_ = _randomized_eigsh(X, n_components=2, selection='module')
+    # eigenvalues
+    assert lambd_.shape == (2,)
+    assert_array_almost_equal(lambd_, [3., -2.])  # negative eigenvalue here
+    # eigenvectors
+    assert alph_.shape == (4, 2)
+
+    # with 'value' selection method, the negative eigenvalue does not show up
+    with pytest.raises(NotImplementedError):
+        _randomized_eigsh(X, n_components=2, selection='value')
+
```

Comment:
Could you please add a bit more tests to compare the results of `_randomized_eigsh` to other `eigsh` implementations on non-diagonal, random yet PSD data?

Reply:
This would indeed be a great addition. I'll see what I can do

Reply:
I added a test, I was surprized by the precision differences but this is maybe related to the nature of the random psd matrix generator.